### ISSUE TITLE
Add a check for the minimum required git version

### ISSUE THREAD

**Issue by hollesse** [remotemobprogramming/mob#332]
As we found out in [#330](https://github.com/remotemobprogramming/mob/pull/330#issuecomment-1277935865) mob can be used with at most git version 2.20.0 and if we ignore the squash-wip feature it can even be used with git 2.13.0. I would like to add a check for the current git version. I am not sure if we should check for 2.13.0 or if we should check for 2.20.0. I would prefere to have a check for mob itself for 2.13.0 and one for the squash-wip feature which should then check for 2.20.0. I would also like to have tests for these version. Therefore we need to check how we could have tests which differs for different git versions. One option to do this could be [go build tags](https://mickey.dev/posts/go-build-tags-testing/)

**Comment by hollesse** [comment]
depends on #330

**Comment by balintf** [comment]
I have started working on this, and I agree with you @hollesse that there should be at least two version checks, one for 2.13.0 for generic feature support, and another for 2.20.0 for squash-wip.
I also have a few questions about how this should work:
- The checks could display a warning using `say.Warning`, continue and eventually fail with another error. Or they could fail after displaying an error with `say.Error`, similarly to how the isGitInstalled check works. Which one should I implement?
- Why is git 2.13.0 the minimum version, what depends on it?
- @hollesse in your first comment you mentioned we could use build tags, did you mean to use them for the testcases for the git version check?
e.g.
```go
// +build git-version-2.12.0
package main
func TestGitVersionTooLow(t *testing.T) {
...
}
```

**Comment by hollesse** [comment]
Regarding 2. we found out that git 2.13.0 is the least supported version. As I remember correctly we had basic tests failing there. See #330. 

Regarding 1. I would therefore fail with an error if we do anything here. This could be added to the guard which checks if git is installed. 

Regarding 3. I meant that we could use the build tags for our CI as we have CI Build running with different Git versions. For the WIP squash part which just works in 2.20.0. But for now I would just do the version check for 2.13.0 as we have an issue to investigate why it didn’t work on 2.19.0. we checked the release notes and do not see any reason for that. I also researched a bit and I think this could really just be a big on our side, but I’m not sure.
